gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -193,7 +194,7 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
         org.apache.gobblin.configuration.State jobStatus = parseJobStatus(gobblinTrackingEvent);
         if (jobStatus != null) {
           try (Timer.Context context = getMetricContext().timer(GET_AND_SET_JOB_STATUS).time()) {
-            addJobStatusToStateStore(jobStatus, this.stateStore, this.eventProducer, this.dagActionStore);
+            addJobStatusToStateStore(jobStatus, this.stateStore, this.eventProducer, this.dagActionStore, this.dagProcEngineEnabled);
```

Comment:
rather than pass one more param to this `static` that exists to be `@VisibleForTesting`, I suggested to pass two fewer params, `this.eventProducer` and `this.dagActionStore`, but to have it return `Optional<JobStatus>` only when the job state went to final.  this would leave the `static` much more cohesive (and testable!) by solely updating the job state store, without leaving side-effects anywhere else